File: python/src/wdl_model/model/resolvers/wdl_import_resolver_filesystem.py
"""Filesystem-only WDL import resolver."""

from __future__ import annotations

from wdl_model.model.errors import WdlImportException

from .wdl_import_resolver_base import WdlImportResolverBase


class WdlImportResolverFilesystem(WdlImportResolverBase):
    """Reject network protocols and only allow file/bare-path imports."""

    def load_http(
        self,
        current_document_location: str | None,
        import_uri: str,
        original_import_location: str,
    ) -> str:
        raise WdlImportException(
            "Filesystem resolver does not support http imports",
            original_import_location,
        )

    def load_https(
        self,
        current_document_location: str | None,
        import_uri: str,
        original_import_location: str,
    ) -> str:
        raise WdlImportException(
            "Filesystem resolver does not support https imports",
            original_import_location,
        )

File: java/src/main/java/com/myriad/wdl/model/resolvers/WdlImportResolverFilesystem.java
package com.myriad.wdl.model.resolvers;

import com.myriad.wdl.model.errors.WdlImportException;
import java.net.URI;

/**
 * Filesystem-only resolver for import locations.
 *
 * <p>Rejects network protocols and delegates filesystem path handling to the base resolver.
 */
public class WdlImportResolverFilesystem extends WdlImportResolverBase {

  @Override
  protected String loadHttp(
      URI currentDocumentLocation, URI importUri, String originalImportLocation)
      throws WdlImportException {
    throw new WdlImportException(
        "Filesystem resolver does not support http imports", originalImportLocation);
  }

  @Override
  protected String loadHttps(
      URI currentDocumentLocation, URI importUri, String originalImportLocation)
      throws WdlImportException {
    throw new WdlImportException(
        "Filesystem resolver does not support https imports", originalImportLocation);
  }
}
